invenio_preservation_sync/models.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-Preservation-Sync is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Models for Preservation Sync integration."""
+
+import uuid
+from enum import Enum
+
+from invenio_db import db
+from invenio_i18n import lazy_gettext as _
+from invenio_webhooks.models import Event
+from sqlalchemy.dialects import mysql, postgresql
+from sqlalchemy_utils.models import Timestamp
+from sqlalchemy_utils.types import ChoiceType, JSONType, UUIDType
+
+PRESERVATION_STATUS_TITLES = {
+    "PRESERVED": _("Preserved"),
+    "PROCESSING": _("Processing"),
+    "FAILED": _("Failed"),
+    "DELETED": _("Deleted"),
+}
+
+PRESERVATION_STATUS_ICON = {
```

Comment:
Why are UI-related fields in the `models.py` (data layer)? Also, the `icon` and `color` are hardcoded here. What happens if we use another icons library or change the colour scheme?

Personally, I would consider moving these closer to the UI and then explicitly matching the status to UI icons/colours there.